roles/os_hardening/tasks/pam_debian.yml

```diff
@@ -0,0 +1,127 @@
+---
+- name: remove pam_cracklib, because it does not play nice with passwdqc
+  apt:
+    name: '{{ os_packages_pam_cracklib }}'
+    state: 'absent'
+  when:
+    - os_auth_pam_passwdqc_enable
+
+- name: install the package for strong password checking
+  apt:
+    name: '{{ os_packages_pam_passwdqc }}'
+    state: 'present'
+    update_cache: 'yes'
+  when:
+    - os_auth_pam_passwdqc_enable
+
+- name: configure passwdqc
+  template:
+    src: 'usr/share/pam-configs/pam_passwdqc.j2'
+    dest: '{{ passwdqc_path }}'
+    mode: '0644'
+    owner: 'root'
+    group: 'root'
+  when:
+    - os_auth_pam_passwdqc_enable
+
+- name: remove passwdqc
+  apt:
+    name: '{{ os_packages_pam_passwdqc }}'
+    state: 'absent'
+  when:
+    - not os_auth_pam_passwdqc_enable
+
+- name: install tally2
+  apt:
+    name: 'libpam-modules'
+    state: 'present'
+  when:
+    - os_auth_retries > 0
+
+- name: manage tally on Debian stable
+  block:
+    - name: configure tally2
+      template:
+        src: 'usr/share/pam-configs/pam_tally2.j2'
+        dest: '{{ tally2_path }}'
+        mode: '0644'
+        owner: 'root'
+        group: 'root'
+      when:
+        - os_auth_retries > 0
+
+    - name: delete tally2 when retries is 0
+      file:
+        path: '{{ tally2_path }}'
+        state: 'absent'
+      when:
+        - os_auth_retries == 0
+  when:
+    - "'libpam-modules' in ansible_facts.packages"
+    - "ansible_facts.packages['libpam-modules'][0].version is version('1.4.0', '<')"
+
+- name: manage tally/faillock on Debian unstable
+  block:
+    - name: delete tally2
+      file:
+        path: '{{ tally2_path }}'
+        state: 'absent'
+
+    - name:  create tally directory
```

Comment:
```suggestion
    - name:  create faillock directory
```

Reply:
technically this directory is refered to as `tally directory` even in faillock. I can change this, in context both names would be correct.